packages/mongodb-artifact-generator/src/commands/generateJiraPromptResponse.ts

```diff
@@ -0,0 +1,413 @@
+        logger.logError(`Error summarizing issue: ${error.item}`);
+        console.log("Error summarizing issue", error.raw);
+      }
+
+      logger.appendArtifact(
+        "summaries.json",
+        JSON.stringify(Object.fromEntries(summariesByIssueKey))
+      );
+
+      // Append summaries to formatted issues
+      const formattedIssuesWithSummaries = formattedJiraIssues.map((issue) => {
+        const summary = summariesByIssueKey.get(issue.key);
+        if (!summary) {
+          throw new Error(`No summary found for issue ${issue.key}`);
+        }
+        return {
+          issue,
+          summary,
+        };
+      }) satisfies FormattedJiraIssueWithSummary[];
+      logger.appendArtifact(
+        "jiraIssues.formattedWithSummaries.json",
+        JSON.stringify(formattedIssuesWithSummaries)
+      );
+
+      // // Generate a list of N questions/prompts for each issue
+      const promptsByIssueKey = new Map<string, string[]>();
+      const generatePrompts = makeGeneratePrompts({
+        openAi: {
+          client: openAiClient,
+          deployment: OPENAI_CHAT_COMPLETION_DEPLOYMENT,
+        },
+        logger,
+        directions: asBulletPoints(
+          "Assume the prompter is not familiar with or looking for the provided Jira issue specifically.",
+          "Assume the prompter is looking for a specific piece of information about the topic or bug discussed in the Jira issue.",
+          "The prompt should be a question that can be answered with the information in the Jira issue but not about the issue itself.",
+          "If the Jira issue is for a specific language, framework, platform, driver, etc., indicate that in the prompt.",
+          "Do not reference hypothetical or speculative information."
+        ),
+      });
+      const { errors: generatePromptsErrors } = await PromisePool.for(
+        formattedIssuesWithSummaries
+      )
+        .withConcurrency(llmMaxConcurrency)
+        .process(async ({ issue, summary }) => {
+          const { prompts } = await generatePrompts({ issue, summary });
+          console.log("generated prompts for issue", issue.key, summary);
+          promptsByIssueKey.set(issue.key, prompts);
+        });
+      for (const error of generatePromptsErrors) {
+        logger.logError(`Error generating prompts: ${error.item}`);
+      }
+      const generatedPrompts: [issueKey: string, prompt: string][] = [
+        ...promptsByIssueKey.entries(),
+      ].flatMap(([issueKey, prompts]) => {
+        return prompts.map((prompt) => [issueKey, prompt] as [string, string]);
+      });
+      logger.appendArtifact(
+        "generatedPrompts.json",
+        JSON.stringify(generatedPrompts)
+      );
+
+      // Have the LLM generate a response to each prompt with the formatted/summarized issue as context
+      const responsesByIssueKey = new Map<
+        string,
+        [prompt: string, response: string][]
+      >();
+
+      const generateResponse = makeGenerateResponse({
```

Comment:
imo this should be evaluated but also for this specific project 🤷

Reply:
https://jira.mongodb.org/browse/EAI-519